backend/packages/Upgrade/src/api/repositories/ExperimentRepository.ts

```diff
@@ -196,17 +269,31 @@ export class ExperimentRepository extends Repository<Experiment> {
         })
       );
 
-    const [experimentData, experimentSegmentData] = await Promise.all([
-      experiment.getMany().catch((errorMsg: any) => {
-        const errorMsgString = repositoryError('ExperimentRepository', 'find', {}, errorMsg);
+    const [experimentConditionLevelPayloadData, experimentFactorPartitionLevelPayloadData, experimentMetricData, experimentSegmentData] = await Promise.all([
+      experimentConditionLevelPayloadQuery.getMany().catch((errorMsg: any) => {
+        const errorMsgString = repositoryError('ExperimentRepository', 'getValidExperiments-experimentConditionLevelPayloadQuery', {}, errorMsg);
         throw errorMsgString;
       }),
-      experimentSegment.getMany().catch((errorMsg: any) => {
-        const errorMsgString = repositoryError('ExperimentRepository', 'find', {}, errorMsg);
+      experimentFactorPartitionLevelPayloadQuery.getMany().catch((errorMsg: any) => {
+        const errorMsgString = repositoryError('ExperimentRepository', 'getValidExperiments-experimentFactorPartitionLevelPayloadQuery', {}, errorMsg);
+        throw errorMsgString;
+      }),
+      experimentMetricQuery.getMany().catch((errorMsg: any) => {
+        const errorMsgString = repositoryError('ExperimentRepository', 'getValidExperiments-experimentMetricQuery', {}, errorMsg);
+        throw errorMsgString;
+      }),
+      experimentSegmentQuery.getMany().catch((errorMsg: any) => {
+        const errorMsgString = repositoryError('ExperimentRepository', 'getValidExperiments-experimentSegmentQuery', {}, errorMsg);
         throw errorMsgString;
       }),
     ]);
 
+    const experimentData = experimentConditionLevelPayloadData.map((data) => {
```

Comment:
this seems like this nested array could be problematic if these arrays are long, maybe it's not important because if it will get cached, but just wanted to think it through. I'm seeing a lot of places where factorial assignments would be doing a lot of nested looping while processing factor / level / payload data.